### ISSUE TITLE
Support for identifying job output under -j

### ISSUE THREAD

**Issue by bdbaddog** [SCons/scons#2584]
This issue was originally created at: 2010-03-18 08:46:48.
This issue was reported by: `ydirson`.
ydirson said at 2010-03-18 08:46:48
>When running in parallel mode and investigating a build failure (possibly due to
>parallelism, which makes non-parallel mode unsuitable for investigation), it
>would be useful to tell which line belongs to which job.
> 
>Some flag which would prefix each line of output with the job number (and
>possibly an aligned "|" as separator) would be of greta help.

garyo said at 2010-04-01 19:12:58
>This is essentially a dup of 1516 (actually a special case of it), which asks
>for a general mechanism for processing command output.  
> 
>*** This issue has been marked as a duplicate of 1516 ***

ydirson said at 2010-04-02 03:21:27
>I can see that #<span></span>1516 can be used to implement this; however, a builtin filter
>implementing this could be also useful (that is, making #<span></span>1516 a dependency of
>this one instead of a dup).

evaned said at 2010-11-15 12:39:44
>Unless I'm missing something, #<span></span>1516 is not quite sufficient to implement the
>feature requested. By the time the SPAWN function is called, the line that
>indicates what command is being run has already been printed (e.g. CCCOMSTR).
> 
>Thus it is possible to use SPAWN to group output lines that were produced by the
>same program but it would take something more to associate them with the command
>that was run.
> 
>(You could print the command being run from within SPAWN, but this has its own
>problems. First, it's obnoxious, because you'll be printing each line twice.
>Second, the actual command may not be what is printed -- the SConscript could
>change CCCOMSTR or whatever. It's probably basically impossible to work
>backwards from what you're given in the SPAWN function to determine what was
>output on the previous line.)

garyo said this issue  is a duplicate of #1516 at 2010-04-01 19:12:58.
